boltons/dictutils.py

```diff
@@ -693,4 +693,104 @@ def __reversed__(self):
             yield curr[KEY]
             curr = curr[PREV]
 
+
+class SentinelInt(int):
+    pass
+
+
+class FrozenDictBase(tuple):
+
+    def __getitem__(self, key):
+        if isinstance(key, SentinelInt):
+            return tuple.__getitem__(self, key)
+        else:
+            try:
+                return getattr(self, str(key.__hash__()))()
+            except AttributeError:
+                raise KeyError(key)
+            except IndexError:
+                raise KeyError(key)
+
+    def __iter__(self):
+        idx = 0
+        while True:
+            try:
+                yield self[SentinelInt(idx)][0]
+            except IndexError:
+                break
+            idx += 1
+
+    def items(self):
+        """Returns a list of tuples of data"""
+        return [(x, self[x]) for x in self]
```

Comment:
also, the implementation here seems to be very inefficient (but since I'm relatively new to Python, I might be missing something). I added this method:

    def alternate_items(self):
        return list(super().__iter__())

And then did that:

    import timeit
    setup = 'from __main__ import frozendict; fd = frozendict({"a": 1, "b": 2, "c": 3})'
    print(min(timeit.repeat('fd.alternate_items()', setup=setup)))  # output: 0.8592938940055319
    print(min(timeit.repeat('fd.items()', setup=setup)))  # output: 10.080946975998813